### PR TITLE
Add CopyQ extension for clipboard management

### PR DESCRIPTION
**Pull request by benedictdudel**
I migrated the old CopyQ extension and added the following features:

- search by value or with copyq entry number
- remove action to remove a specific clipboard item

Let me know what you think!

### CODE REVIEW

File: CopyQ.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+import subprocess
+from albertv0 import *
+from shutil import which
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "CopyQ"
+__version__ = "1.0"
+__trigger__ = "cp "
+__author__ = "Benedict Dudel"
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: if its really based on the copyq you should mention the author.

File: CopyQ.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
+""" CopyQ Clipboard Management """
+
+import subprocess
+from albertv0 import *
+from shutil import which
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "CopyQ"
+__version__ = "1.0"
+__trigger__ = "cp "
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: use cq please

File: CopyQ.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+if which("copyq") is None:
+    raise Exception("'copyq' is not in $PATH.")
+
+iconPath = iconLookup('copyq')
+
+def handleQuery(query):
+    if query.isTriggered:
+        if not query.string or query.string == "":
+            print("show last items")
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: remove print. there is not stdout (conceptually) use the [logging functions](https://albertlauncher.github.io/docs/extensions/python/#free-functions) for that

File: CopyQ.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            return getLastItems()
+
+        item = searchItems(query.string)
+        if not item == None:
+            print("show search item")
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: here as well

**Comment by ManuelSchneid3r** [comment]
@BarbUk are your latest changes in here? Can you please take a look at it?

File: CopyQ.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+import subprocess
+from albertv0 import *
+from shutil import which
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "CopyQ"
+__version__ = "1.0"
+__trigger__ = "cq "
+__author__ = "Lukas Holecek"
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: Who is Lukas? I didnt want to remove your name :D I just wanted to keep credits for former contributors. Just add your name to the string, separated by comma.

Comment (ManuelSchneid3r) [reply CopyQ.py]: Ah, now I see. Lukas is the author of copyq. The __autor__ variable is intended to specify the author[s] of the extension, i.e. you and BarbUk .

Comment (benedictdudel) [reply CopyQ.py]: Ah, I now understand :P

File: CopyQ.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            icon=iconPath,
+            text=__prettyname__,
+            subtext="No entry found for your search query.",
+            completion="",
+            actions=[]
+    )]
+
+def getLastItems():
+    size = getStackSize()
+    if size >= 14:
```
Comment (raghuramos1987) [inline CopyQ.py]: why 14 though?

File: CopyQ.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    item = searchItemByItemNumber(queryValue.strip())
+    if not item == None:
+        return [item]
+
+    return None
+
+def searchItemByValue(queryValue):
+    size = getStackSize()
+    if size >= 50:
```
Comment (raghuramos1987) [inline CopyQ.py]: why 50 specifically?

File: CopyQ.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+def getAlbertItem(id, text, subtext):
+    text = getStrippedClipboardText(text)
+    if not text:
+        return None
+
+    return Item(
+            id="%s" % id,
+            icon=iconPath,
+            text="%s" % text,
+            subtext="%s: %s" % (id, text),
```
Comment (raghuramos1987) [inline CopyQ.py]: why are you adding the id?

**Comment by ManuelSchneid3r** [comment]
I locally made a new version. Merging the best of all. I will release it with the next version.

**Pull request closed without merging**